tests/unit/s2n_self_talk_inet_socket_test.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <fcntl.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#define S2N_TEST_INADDR_LOOPBACK 0x7f000001 /* 127.0.0.1 */
+
+#define WRITE_sync()                                             \
+    EXPECT_SUCCESS(sleep(1)); /* allow time for data to flush */ \
+    EXPECT_SUCCESS(write(write_pipe, &sync, 1));                 \
+    EXPECT_SUCCESS(sleep(1)); /* allow reader time time process current operation */
+
+#define READ_sync() \
+    EXPECT_SUCCESS(read(read_pipe, &sync, 1));
+
+#define SEND_CHAR(conn, c) \
+    send_buffer[0] = c;    \
+    EXPECT_SUCCESS(s2n_send(conn, send_buffer, 1, &blocked));
```

Comment:
implicit variables in macros is pretty annoying to work with. why can't these just be functions?